demos/guide/tls-server-block.c

```diff
@@ -0,0 +1,286 @@
+/*
+ *  Copyright 2024 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ *  Licensed under the Apache License 2.0 (the "License").  You may not use
+ *  this file except in compliance with the License.  You can obtain a copy
+ *  in the file LICENSE in the source distribution or at
+ *  https://www.openssl.org/source/license.html
+ */
+
+/*
+ * NB: Changes to this file should also be reflected in
+ * doc/man7/ossl-guide-tls-server-block.pod
+ */
+
+#include <string.h>
+
+/* Include the appropriate header file for SOCK_STREAM */
+#ifdef _WIN32 /* Windows */
+# include <stdarg.h>
+# include <winsock2.h>
+#else /* Linux/Unix */
+# include <err.h>
+# include <sys/socket.h>
+# include <sys/select.h>
+#endif
+
+#include <openssl/bio.h>
+#include <openssl/ssl.h>
+#include <openssl/err.h>
+
+static const char cache_id[] = "OpenSSL Demo Server";
+
+#ifdef _WIN32
+static const char *progname;
+
+static void vwarnx(const char *fmt, va_list ap)
+{
+    if (progname != NULL) {
+        fprintf(stderr, "%s: ", progname);
+    }
+    vfprintf(stderr, fmt, ap);
+    putc('\n', stderr);
+}
+
+static void errx(int status, const char *fmt, ...)
+{
+    va_list ap;
+    va_start(ap, fmt);
+    vwarnx(fmt, ap);
+    va_end(ap);
+    exit(status);
+}
+
+static void warnx(const char *fmt, ...)
+{
+    va_list ap;
+    va_start(ap, fmt);
+    vwarnx(fmt, ap);
+    va_end(ap);
+}
+#endif
+
+/* Minimal TLS echo server. */
+int main(int argc, char *argv[])
+{
+    int res = EXIT_FAILURE;
+    long opts;
+    const char *hostport;
+    SSL_CTX *ctx = NULL;
+    BIO *acceptor_bio;
+
+#ifdef _WIN32
+    progname = argv[0];
+#endif
+
+    if (argc != 2) {
+        errx(res, "Usage: %s [host:]port", argv[0]);
+    }
```

Comment:
nit: `{}` not necessary for single line if

Reply:
Done.